File: DSA/Deeptansh-nagarr/day 10/question2.java
import java.util.Scanner;
import java.util.Stack;

class QueueUsingTwoStacks {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public QueueUsingTwoStacks() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }

    public void enqueue(int element) {
        stack1.push(element);
    }

    public int dequeue() {
        if (stack2.isEmpty()) {
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
        }
        return stack2.pop();
    }

    public static void main(String[] args) {
        QueueUsingTwoStacks queue = new QueueUsingTwoStacks();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("1. Enqueue");
            System.out.println("2. Dequeue");
            System.out.println("3. Exit");
            System.out.print("Choose an option: ");
            int option = scanner.nextInt();

            switch (option) {
                case 1:
                    System.out.print("Enter an element to enqueue: ");
                    int element = scanner.nextInt();
                    queue.enqueue(element);
                    break;
                case 2:
                    try {
                        System.out.println("Dequeued element: " + queue.dequeue());
                    } catch (Exception e) {
                        System.out.println("Queue is empty");
                    }
                    break;
                case 3:
                    System.exit(0);
                    break;
                default:
                    System.out.println("Invalid option");
            }
        }
    }
}

File: DSA/Advik/Day10/question2.cpp
#include <iostream>
#include <stack>
using namespace std;
class Queue {
private:
    std::stack<int> stack1;
    std::stack<int> stack2; 
    void transferStack1ToStack2() {
        while (!stack1.empty()) {
            stack2.push(stack1.top());
            stack1.pop();
        }
    }
public:
    void enqueue(int value) {
        stack1.push(value);
    }
    int dequeue() {
        if (stack2.empty()) {
            if (stack1.empty()) {
                cerr << "Queue is empty, cannot dequeue." << endl;
                exit(EXIT_FAILURE);
            }
            transferStack1ToStack2();
        }
        int front = stack2.top();
        stack2.pop();
        return front;
    }
    bool isEmpty() {
        return stack1.empty() && stack2.empty();
    }
    int front() {
        if (stack2.empty()) {
            if (stack1.empty()) {
                cerr << "Queue is empty, no front element." << endl;
                exit(EXIT_FAILURE);
            }
            transferStack1ToStack2();
        }
        return stack2.top();
    }
};
int main() {
    Queue q;
    q.enqueue(1);
    q.enqueue(2);
    q.enqueue(3);
    q.enqueue(4);
    q.enqueue(5);
    cout << "Dequeued: " << q.dequeue() << endl;
    cout << "Dequeued: " << q.dequeue() << endl;
    q.enqueue(6);
    q.enqueue(7);
    while (!q.isEmpty()) {
        cout << "Dequeued: " << q.dequeue() << endl;
    }
    return 0;
}